tools/run-dev

```diff
@@ -66,6 +66,11 @@ parser.add_argument(
 parser.add_argument(
     "--skip-help-center-build", action="store_true", help="Skip the build step for help center"
 )
+parser.add_argument(
+    "--help-center-dev-server",
+    action="store_true",
+    help="Run dev server for help center. Hot reload will work for this mode, but search will not work in the generated website.",
+)
```

Comment:
I think probably the option structure for this detail isn't what we want.
- The default should be hot-reload, because that's what most developers will want for regular work.
- If we need an option to disable hot-reload in order to make search work, we could do that, but I'm not entirely sure why that would be necessary -- presumably the hot-reloadable dev server is compatible with search (but doesn't update the search index), and we could just have instructions of the form "If you'd like to test help center search, run this command to build the search index"?

Reply:
>  presumably the hot-reloadable dev server is compatible with search (but doesn't update the search index), and we could just have instructions of the form "If you'd like to test help center search, run this command to build the search index"?

This should technically be possible and we can look into providing this option (preferably) upstream on starlight or on our side with some modification of the existing search component. But I think we can do it as a follow-up. Until then we can have 2 flags, one for dev server and one for the production build to be served via aiohttp

Reply:
Yeah that seems fine.